From: =?utf-8?q?Jonas_=C3=85dahl?= <jadahl@gmail.com>
Date: Wed, 10 Sep 2025 12:09:01 +0200
Subject: wayland/xdg-shell: Make invalid geometry warning more descriptive

It'll contain app id and window title, if available.

Bug: https://gitlab.gnome.org/GNOME/mutter/-/issues/4310
Forwarded: https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/4648
---
 src/wayland/meta-wayland-xdg-shell.c | 13 +++++++++----
 1 file changed, 9 insertions(+), 4 deletions(-)

diff --git a/src/wayland/meta-wayland-xdg-shell.c b/src/wayland/meta-wayland-xdg-shell.c
index 65350b9..ac3b675 100644
--- a/src/wayland/meta-wayland-xdg-shell.c
+++ b/src/wayland/meta-wayland-xdg-shell.c
@@ -1994,10 +1994,15 @@ meta_wayland_xdg_surface_post_apply_state (MetaWaylandSurfaceRole  *surface_role
                                                      &priv->geometry);
       if (priv->geometry.width == 0 || priv->geometry.height == 0)
         {
-          g_warning ("Invalid window geometry for xdg_surface@%d. Ignoring "
-                     "for now, but this will result in client termination "
-                     "in the future.",
-                     wl_resource_get_id (priv->resource));
+          MetaWaylandSurface *surface =
+            meta_wayland_surface_role_get_surface (surface_role);
+          MetaWindow *window = meta_wayland_surface_get_window (surface);
+
+          g_warning ("Client provided invalid window geometry for "
+                     "xdg_surface#%d (%s - %s). Working around.",
+                     wl_resource_get_id (priv->resource),
+                     window ? window->res_class : "N\\A",
+                     window ? window->desc : "N\\A");
           return;
         }
 
